src/FSharp.Configuration/TypeProviders.Helper.fs

```diff
@@ -229,6 +229,9 @@ module File =
                     | _, content -> return content 
                 finally file.Dispose() 
             | None ->  
+                if attemptsLeft = 0 
+                    then return raise (FileNotFoundException(sprintf "File could not be opened after %d attempts." maxAttempts))
```

Comment:
What about adding the file name into the exception message? I'd make it much more informative.